### PR TITLE
Fix ##4522: Removes canvas rule on fingerprint protection

### LINKED ISSUE #4522: Remove canvas fingerprinting protection
Currently, the canvas fingerprinting protection is breaking captchas for iOS users. Remove the canvas fingerprint protection for now and improve the fingerprint protection implementation with farbling etc later on.

test plan https://github.com/brave/internal/issues/844

### PR DESCRIPTION
**Pull request by gpestana**
## Summary of Changes

This pull request resolves https://github.com/brave/brave-ios/issues/4522

test plan https://github.com/brave/internal/issues/844

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Comment by gpestana** [comment]
cc @iccub

**Review by iccub — APPROVED** [review]

File: Client/Frontend/UserContent/UserScripts/FingerprintingProtection.js [hunk 1]
```diff
@@ -13,15 +13,7 @@ if (webkit.messageHandlers.$<handler>) {
...
-                    // Prevent fingerprinting using contentDocument.createElement('canvas'),
-                    // which evades trapInstanceMethod when the iframe is sandboxed
-                    sendMessage({ obj: `${obj}`, method: method })
-                } else {
-                    // Otherwise apply the original method
-                    return orig.apply(this, arguments)
-                }
+                return orig.apply(this, arguments)
             };
             return orig;
```
Comment (kylehickinson) [inline Client/Frontend/UserContent/UserScripts/FingerprintingProtection.js]: This only removes the protection on making a `canvas` element using `createElement`/`createElementNS`.  In fact if thats all thats needed this whole function can be removed and the hooks removed where `installIFrameMethodTrap` is called.

There are two other `<canvas>` element method hooks in place (`toBlob` and `toDataURL`), do these also need to be removed?

**Review by kylehickinson — CHANGES_REQUESTED** [review]

File: Client/Frontend/UserContent/UserScripts/FingerprintingProtection.js (hunk 1 above)
Comment (gpestana) [reply Client/Frontend/UserContent/UserScripts/FingerprintingProtection.js]: Good catch @kylehickinson! I just updated the PR removing both `toBlob` and `toDataURL` hooks.

**Review by Brandon-T — APPROVED** [review]

File: Client/Frontend/UserContent/UserScripts/FingerprintingProtection.js [hunk 2]
```diff
@@ -13,15 +13,7 @@ if (webkit.messageHandlers.$<handler>) {
...
-                var last = arguments[arguments.length - 1]
-                if (last && last.toLowerCase() === 'canvas') {
-                    // Prevent fingerprinting using contentDocument.createElement('canvas'),
-                    // which evades trapInstanceMethod when the iframe is sandboxed
-                    sendMessage({ obj: `${obj}`, method: method })
-                } else {
-                    // Otherwise apply the original method
-                    return orig.apply(this, arguments)
-                }
+                return orig.apply(this, arguments)
```
Comment (Brandon-T) [inline Client/Frontend/UserContent/UserScripts/FingerprintingProtection.js]: You can remove this entire hook. All it is doing is calling the original method.

**Review by Brandon-T — CHANGES_REQUESTED** [review]

File: Client/Frontend/UserContent/UserScripts/FingerprintingProtection.js (hunk 2 above)
Comment (gpestana) [reply Client/Frontend/UserContent/UserScripts/FingerprintingProtection.js]: @Brandon-T thanks for the heads up -- I removed the `installIFrameMethodTrap` altogether.

**Review by kylehickinson — APPROVED** [review]

**Review by Brandon-T — APPROVED** [review]

**Pull request merged**